File: fudosinServer/src/main/java/com/example/demo/entity/Person.java
package com.example.demo.entity;


import javax.persistence.*;
import java.util.Date;
import java.util.Set;

@Entity
@Table
//@MappedSuperclass
public class Person {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String firstName;
    private String lastName;
    private String SecondName;
    private Date birthday;
    private String email;
    private String telephone;
    private String username;
    private String password;


    @ManyToOne
    @JoinColumn
    private Person parent;

    public Person() {
    }

    @ManyToMany(fetch = FetchType.LAZY)
    @JoinTable(name = "user_roles",
            joinColumns = @JoinColumn(name = "user_id"),
            inverseJoinColumns = @JoinColumn(name = "role_id"))
    private Set<Role> userRoles;

    @ManyToMany(fetch = FetchType.LAZY)
    @JoinTable(name = "user_group",
            joinColumns = @JoinColumn(name = "user_id"),
            inverseJoinColumns = @JoinColumn(name = "group_id"))
    private Set<TrainingGroup> userGroup;

    @OneToMany
    private Set<Visit> visits;

    public Person(String username, String email, String password) {
        this.username = username;
        this.email = email;
        this.password = password;
    }

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getFirstName() {
        return firstName;
    }

    public void setFirstName(String firstName) {
        this.firstName = firstName;
    }

    public String getLastName() {
        return lastName;
    }

    public void setLastName(String lastName) {
        this.lastName = lastName;
    }

    public String getSecondName() {
        return SecondName;
    }

    public void setSecondName(String secondName) {
        SecondName = secondName;
    }

    public Date getBirthday() {
        return birthday;
    }

    public void setBirthday(Date birthday) {
        this.birthday = birthday;
    }

    public String getEmail() {
        return email;
    }

    public void setEmail(String email) {
        this.email = email;
    }

    public String getTelephone() {
        return telephone;
    }

    public void setTelephone(String telephone) {
        this.telephone = telephone;
    }

    public Set<Role> getUserRoles() {
        return userRoles;
    }

    public void setUserRoles(Set<Role> userRoles) {
        this.userRoles = userRoles;
    }

    public Set<TrainingGroup> getUserGroup() {
        return userGroup;
    }

    public void setUserGroup(Set<TrainingGroup> userGroup) {
        this.userGroup = userGroup;
    }

    public Set<Visit> getVisits() {
        return visits;
    }

    public void setVisits(Set<Visit> visits) {
        this.visits = visits;
    }

    public String getUsername() {
        return username;
    }

    public void setUsername(String username) {
        this.username = username;
    }

    public Person getParent() {
        return parent;
    }

    public void setParent(Person parent) {
        this.parent = parent;
    }

    public String getPassword() {
        return password;
    }

    public void setPassword(String password) {
        this.password = password;
    }
}

File: fudosinClient/src/app/entity/Person.ts
import {Role} from "./Role";

export class Person {
  private _id: number | undefined;
  private _firstName: string | undefined;
  private _secondName: string | undefined;
  private _LastName: string | undefined;
  private _telephone: number | undefined;
  private _parentId: Person | undefined;
  private _email: string | undefined;
  private _birthDay: Date | undefined;
  private _roles: Role[] | undefined;
  private _userPassword: string;


  constructor() {
    this._userPassword = "1234";

  }

  get id(): number | undefined {
    return this._id;
  }

  set id(value: number | undefined) {
    this._id = value;
  }

  get firstName(): string | undefined {
    return this._firstName;
  }

  set firstName(value: string | undefined) {
    this._firstName = value;
  }

  get secondName(): string | undefined {
    return this._secondName;
  }

  set secondName(value: string | undefined) {
    this._secondName = value;
  }

  get LastName(): string | undefined {
    return this._LastName;
  }

  set LastName(value: string | undefined) {
    this._LastName = value;
  }

  get telephone(): number | undefined {
    return this._telephone;
  }

  set telephone(value: number | undefined) {
    this._telephone = value;
  }

  get parentId(): Person | undefined {
    return this._parentId;
  }

  set parentId(value: Person | undefined) {
    this._parentId = value;
  }

  get email(): string | undefined {
    return this._email;
  }

  set email(value: string | undefined) {
    this._email = value;
  }

  get birthDay(): Date | undefined {
    return this._birthDay;
  }

  set birthDay(value: Date | undefined) {
    this._birthDay = value;
  }


  get roles(): Role[] | undefined {
    return this._roles;
  }

  set roles(value: Role[] | undefined) {
    this._roles = value;
  }

  get userPassword(): string {
    return this._userPassword;
  }

  set userPassword(value: string) {
    this._userPassword = value;
  }
}
